docs/eessi/high-level-design.md

```diff
@@ -0,0 +1,241 @@
+
+$ ls -l /cvmfs/software.eessi.io/versions/2023.06/compat/linux/x86_64/lib64
+total 4923
+-rwxr-xr-x 1 cvmfs cvmfs  210528 Nov 15 11:22 ld-linux-x86-64.so.2
+...
+-rwxr-xr-x 1 cvmfs cvmfs 1876824 Nov 15 11:22 libc.so.6
+...
+-rwxr-xr-x 1 cvmfs cvmfs  911600 Nov 15 11:22 libm.so.6
+...
+```
+
+Libraries included in the compatibility layer can be used on any Linux client system,
+as long as the CPU family is compatible and taken into account.
+
+```
+$ uname -m
+x86_64
+
+$ cat /etc/redhat-release
+Red Hat Enterprise Linux release 8.8 (Ootpa)
+
+$ /cvmfs/software.eessi.io/versions/2023.06/compat/linux/x86_64/lib64/libc.so.6
+GNU C Library (Gentoo 2.37-r7 (patchset 10)) stable release version 2.37.
+...
+```
+
+By making sure that the software installations included in EESSI **only rely on tools and libraries
+provided by the compatibility layer**, and do not (directly) require anything from the client OS,
+we can ensure that they can be used in a broad variety of Linux systems,
+regardless of the (version of) Linux distribution being used.
+
+!!! note
+
+    This is very similar to the OS tools and libraries that are included in container images,
+    except that no container runtime is involved here (typically), only CernVM-FS.
+
+### Software layer {: #software_layer }
+
+The top layer of EESSI is called the *software layer*, which contains the actual scientific software applications
+and their dependencies.
+
+#### EasyBuild to install software {: #easybuild }
+
+Building, managing, and optimising the software installations included in the software layer is
+layer is done using [**EasyBuild**](https://easybuild.io), a well-established software build and installation framework
+for managing (scientific) software stacks on High-Performance Computing (HPC) systems.
+
+#### Lmod as user interface {: #lmod }
+
+Next to installing the software itself, EasyBuild also automatically generates [*environment module
+files*](https://en.wikipedia.org/wiki/Environment_Modules_(software)).
+These files, which are essentially small Lua scripts, are consumed via [**Lmod**](https://lmod.readthedocs.io),
+a modern implementation of the concept of environment modules which provides a user-friendly interface to end users of EESSI.
+
+#### CPU detection via archspec {: #cpu_detection }
+
+The [**`archspec`**](https://archspec.readthedocs.io) Python library is leveraged by
+an initialisation script that is included in the EESSI repository to detect the CPU family and microarchitecture of
+a client system. Based on the supported CPU features, the EESSI initialisation
+script will **automatically select the best suited subdirectory** of the software layer that contains
+software installations that are optimised for that particular type of CPU, and update the session environment
+to start using it.
+
+A more lightweight pure bash alternative to `archspec` named `archdetect` is being evaluated
+to become part of the initialisation procedure (and replace `archspec`).
```

Comment:
I've reworded this section a bit to put `archspec` and `archdetect` at the same level (since we're currently using `archdetect` as default mechanism in `software.eessi.io`)